Cargo.toml

```diff
@@ -21,3 +22,68 @@ members = [
     "examples/concurrent",
     "examples/firehose",
 ]
+
+[workspace.package]
+edition = "2021"
+rust-version = "1.70"
+repository = "https://github.com/sugyan/atrium"
+license = "MIT"
+keywords = ["atproto", "bluesky"]
+
+[workspace.dependencies]
+# Intra-workspace dependencies
+atrium-api = { version = "0.16.0", path = "atrium-api" }
+atrium-codegen = { version = "0.1.0", path = "atrium-codegen" }
+atrium-lex = { version = "0.1.0", path = "atrium-lex" }
+atrium-xrpc = { version = "0.8.0", path = "atrium-xrpc" }
+atrium-xrpc-client = { version = "0.2.0", path = "atrium-xrpc-client" }
+atrium-xrpc-server = { version = "0.1.0", path = "atrium-xrpc-server" }
+
+# async in traits
+# Can be removed once MSRV is at least 1.75.0.
+async-trait = "0.1.68"
+
+# DAG-CBOR codec and CAR format
+ciborium = "0.2.1"
+cid = "0.10.1"
+libipld-core = "0.16"
+rs-car = "0.4.1"
+serde_ipld_dagcbor = "0.3"
+
+# Serde
+serde = "1.0.160"
+serde_bytes = "0.11.9"
+serde_json = "1.0.96"
+serde_qs = "0.12"
+serde_repr = "0.1.12"
+serde_with = "2.3.2"
+
+# Networking
+futures = { version = "0.3.28", default-features = false }
```

Comment:
Since `features` seems to be used only in `atrium-xrpc-client` with `default-features = false`, it seems better to set `default-features = false` only in `atrium-xrpc-client` without disabling it here.

Reply:
Feature flags are always additive, and the features declared in the `[workspace.dependencies]` are merged with the features declared in a given crate. Thus when using workspace inheritance like this, you can't set `default-features = false` at a per-crate level, because they are being default-enabled at the workspace level and the setting would be ignored. For this reason, [`cargo` does not allow you to use the `default-features` key at the crate level for inherited dependencies](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#inheriting-a-dependency-from-a-workspace).

Reply:
I see. Thank you very much.
Then I guess it can be set as `default-features = true` (or remove specifying about features) for the workspace dependencies. In `atrium-xrpc-client`, `futures` is just used as `dev-dependencies` for tests, so it won't affect the published crates, right?

Reply:
Correct, the published crates only depend on what are declared in their own dependency blocks (with any features that are enabled in the workspace, merged with any features in their block).